django_socio_grpc/servicer_proxy.py

```diff
@@ -1,130 +1,252 @@
-                    instance_action = getattr(service_instance, action)
-                    if asyncio.iscoroutinefunction(instance_action):
-                        instance_action = async_to_sync(instance_action)
-                    return instance_action(service_instance.request, service_instance.context)
-                except GRPCException as grpc_error:
-                    logger.error(grpc_error)
-                    context.abort(grpc_error.status_code, grpc_error.get_full_details())
-                except Exception as error:
-                    etype, value, tb = sys.exc_info()
-                    formatted_exception = traceback.format_exception(etype, value, tb)
-                    # No need to send it to µservices logging because we did it as exception with log_unhandled_exception
-                    logger.error("".join(formatted_exception), extra={"emit_to_server": False})
-                    grpcHandler = GRPCHandler()
-                    grpcHandler.log_unhandled_exception(etype, value, tb)
-                    context.abort(grpc.StatusCode.UNKNOWN, str(error))
-                finally:
-                    # INFO - AM - 30/06/2021 - Need this in production environnement to avoid SSL end of files errors when too much connection on database
-                    close_old_connections()
-                    pass
-
-            return handler
-
-    def create_service(self):
-        return self.service_class(**self.initkwargs)
+    def create_service(self, **kwargs):
+        return self.service_class(**self.initkwargs, **kwargs)
 
     def __getattr__(self, action):
         if not hasattr(self.service_class, action):
             raise Unimplemented()
 
-        return self.call_handler(action)
+        return self.get_handler(action)
 
-
-def close_old_connections():
-    for conn in db.connections.all():
-        if conn.connection is None:
-            continue
-        if conn.get_autocommit():
-            conn.close_if_unusable_or_obsolete()
+    def process_exception(self, exc, request: GRPCRequestContainer):
+        if isinstance(exc, GRPCException):
+            logger.error(exc)
+            request.context.abort(exc.status_code, exc.get_full_details())
+        elif isinstance(exc, grpc.RpcError):
+            raise exc
+        else:
+            etype, value, tb = sys.exc_info()
+            formatted_exception = traceback.format_exception(etype, value, tb)
+            # No need to send it to µservices logging because we did it as exception with log_unhandled_exception
+            logger.error("".join(formatted_exception), extra={"emit_to_server": False})
+            grpc_handler = GRPCHandler()
+            grpc_handler.log_unhandled_exception(etype, value, tb)
+            request.context.abort(grpc.StatusCode.UNKNOWN, str(exc))
+
+    async def async_process_exception(self, exc, request: GRPCRequestContainer):
```

Comment:
why not callin process_exception with sync_to_async ? or reverse to avoid duplicate code

Reply:
The `context.abort` method will either be async or not depending on the running mode